File: src/frontend/src/models/bed.ts
import { ActuatorRef } from "./actuator";
import { AutomationRule } from "./automation";
import { PlantRef } from "./plant";
import { SensorRef } from "./sensor";

export type BedRef = {
  id: string;
};

export type Bed = BedRef & {
  name: string;
  description: string;
  plant: PlantRef;
  sensors: SensorRef[];
  actuators: ActuatorRef[];
  rules: AutomationRule[];
};

File: src/backend/SmartGarden.EntityFramework/Models/Bed.cs
using System.ComponentModel.DataAnnotations.Schema;
using SmartGarden.EntityFramework.Core.Models;
using SmartGarden.Modules.Enums;

namespace SmartGarden.EntityFramework.Models;

public class Bed : BaseEntity
{
    public string Name { get; set; }
    public string Description { get; set; }


    [ForeignKey(nameof(Plant))]
    public Guid PlantId { get; set; }
    public virtual Plant Plant { get; set; }

    public virtual List<ModuleRef> Modules { get; set; } = new();
    public virtual List<AutomationRule> Rules { get; set; } = new();

    [NotMapped]
    public virtual IEnumerable<ModuleRef> Actuators => Modules
        .Where(m => m.Type.IsActuator());
   
    [NotMapped]
    public virtual IEnumerable<ModuleRef> Sensors => Modules
        .Where(m => m.Type.IsSensor());
        
}